### PR TITLE
fix: remove GCP Config and GCR logic

### PR DESCRIPTION
**Pull request by rmoles**
fixes https://lacework.atlassian.net/browse/ALLY-417 & https://lacework.atlassian.net/browse/ALLY-416
Signed-off-by: Ross Moles <ross.moles@lacework.net>

### CODE REVIEW

**Comment by rmoles** [comment]
Confirmed with all vars set to default that this plan runs successfully:
```
terraform apply
random_id.uniq: Refreshing state... [id=dWWICg]
data.google_project.selected: Refreshing state...

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_service_account.lacework[0] will be created
  + resource "google_service_account" "lacework" {
      + account_id   = "lwsvc-7565880a"
      + display_name = "lwsvc-7565880a"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = "lacework-ecosystem2"
      + unique_id    = (known after apply)
    }

  # google_service_account_key.lacework[0] will be created
  + resource "google_service_account_key" "lacework" {
      + id                 = (known after apply)
      + key_algorithm      = "KEY_ALG_RSA_2048"
      + name               = (known after apply)
      + private_key        = (sensitive value)
      + private_key_type   = "TYPE_GOOGLE_CREDENTIALS_FILE"
      + public_key         = (known after apply)
      + public_key_type    = "TYPE_X509_PEM_FILE"
      + service_account_id = (known after apply)
      + valid_after        = (known after apply)
      + valid_before       = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_service_account.lacework[0]: Creating...
google_service_account.lacework[0]: Creation complete after 2s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-7565880a@lacework-ecosystem2.iam.gserviceaccount.com]
google_service_account_key.lacework[0]: Creating...
google_service_account_key.lacework[0]: Creation complete after 2s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-7565880a@lacework-ecosystem2.iam.gserviceaccount.com/keys/2424429b9190986b85b79e28ba27de882700e3eb]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

created = true
name =
private_key = .....
project_id = lacework-ecosystem2
```

**Comment by rmoles** [comment]
```
 terraform apply
data.google_project.selected: Refreshing state...

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_service_account.lacework[0] will be created
  + resource "google_service_account" "lacework" {
      + account_id   = (known after apply)
      + display_name = (known after apply)
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = "lacework-ecosystem2"
      + unique_id    = (known after apply)
    }

  # google_service_account_key.lacework[0] will be created
  + resource "google_service_account_key" "lacework" {
      + id                 = (known after apply)
      + key_algorithm      = "KEY_ALG_RSA_2048"
      + name               = (known after apply)
      + private_key        = (sensitive value)
      + private_key_type   = "TYPE_GOOGLE_CREDENTIALS_FILE"
      + public_key         = (known after apply)
      + public_key_type    = "TYPE_X509_PEM_FILE"
      + service_account_id = (known after apply)
      + valid_after        = (known after apply)
      + valid_before       = (known after apply)
    }

  # random_id.uniq will be created
  + resource "random_id" "uniq" {
      + b64_std     = (known after apply)
      + b64_url     = (known after apply)
      + byte_length = 4
      + dec         = (known after apply)
      + hex         = (known after apply)
      + id          = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

random_id.uniq: Creating...
random_id.uniq: Creation complete after 0s [id=gHDZKQ]
google_service_account.lacework[0]: Creating...
google_service_account.lacework[0]: Creation complete after 3s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-8070d929@lacework-ecosystem2.iam.gserviceaccount.com]
google_service_account_key.lacework[0]: Creating...
google_service_account_key.lacework[0]: Creation complete after 2s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-8070d929@lacework-ecosystem2.iam.gserviceaccount.com/keys/dc499ee8eac1ee9b6704095f159f243ce0bb75c5]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

created = true
name =
private_key = ......
project_id = lacework-ecosystem2
```

**Comment by afiune** [comment]
@rmoles really quick, why the output of that `terraform apply` did not show the `name` in the outputs?

File: examples/custom-service-account/README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Create Service Account With Custom Settings
 
-This example shows how to create a service account to provide Lacework read-only access to a Google Cloud Organization and customizing the account with a few inputs.
+This example shows how to create a service account to provide 
+Lacework read-only access to a Google Cloud project, customizing 
+the account with a few inputs.
```
Comment (afiune) [inline examples/custom-service-account/README.md]: Does this new module actually provide read-only access? or it only creates an empty service account? 🤔

File: examples/custom-service-account/README.md [hunk 2]
```diff
@@ -22,8 +24,6 @@ provider "lacework" {}
 module "lacework_svc_account" {
   source               = "lacework/service-account/gcp"
   version              = "~> 0.1.0"
```
Comment (afiune) [inline examples/custom-service-account/README.md]: All versions should be prepared for the major version bump we are going to do:
```suggestion
  version              = "~> 1.0"
```

Comment (afiune) [reply examples/custom-service-account/README.md]: For doubts about why we are following this new version pin, read https://lacework.atlassian.net/wiki/spaces/ALLY/pages/1844511095/Semantic+Versioning

File: examples/custom-service-account/README.md (hunk 1 above)
Comment (rmoles) [reply examples/custom-service-account/README.md]: good catch. It creates an empty service account with no roles/permissions. I'll update this.

**Comment by rmoles** [comment]
> @rmoles really quick, why the output of that `terraform apply` did not show the `name` in the outputs?

I think this is pointing at the var rather than the local. Let me test and update.

**Comment by rmoles** [comment]
@afiune the name now shows in the apply
```
random_id.uniq: Creating...
random_id.uniq: Creation complete after 0s [id=Co3cHQ]
google_service_account.lacework[0]: Creating...
google_service_account.lacework[0]: Creation complete after 4s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-0a8ddc1d@lacework-ecosystem2.iam.gserviceaccount.com]
google_service_account_key.lacework[0]: Creating...
google_service_account_key.lacework[0]: Creation complete after 3s [id=projects/lacework-ecosystem2/serviceAccounts/lwsvc-0a8ddc1d@lacework-ecosystem2.iam.gserviceaccount.com/keys/cf1ddeaa3c3ea003522742d4bcf9203d979a18ab]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

created = true
name = lwsvc-0a8ddc1d
private_key = .......
```

File: output.tf [hunk 3]
```diff
@@ -4,7 +4,7 @@ output "created" {
 }
 
 output "name" {
-  value       = var.service_account_name
+  value       = local.service_account_name
```
Comment (afiune) [inline output.tf:7]: Nice

File: examples/default-service-account/README.md [hunk 4]
```diff
@@ -1,7 +1,8 @@
 # Create Service Account With Default Settings
 
-This example shows how to create a service account to provide Lacework
-read-only access to a Google Cloud Project.
+This example shows how to create a service account with no roles
```
Comment (afiune) [inline examples/default-service-account/README.md:3]: cool

**Review by afiune — APPROVED** [review]
tenor-66544608

**Comment by afiune** [comment]
since we are modifying the service-account module, it is redundant to add that into the commit message, I renamed it.

**Pull request merged**